src/components/SearchBar.jsx

```diff
@@ -0,0 +1,187 @@
+import React, { Component } from 'react'
+import { translate } from '../lib/I18n'
+import Autosuggest from 'react-autosuggest';
+import { fetchRawIntent } from '../lib/intents'
+
+const INTENT_VERB = 'OPEN'
+const INTENT_DOCTYPE = 'io.cozy.suggestions'
+
+class SearchBar extends Component {
+  constructor (props) {
+    super(props)
+    
+    this.state = {
+      query: '',
+      suggestions: [],
+      sourceURLs: []
+    }
+    
+    this.sources = []
+  }
+  
+  componentDidMount () {
+    // The launcher has one or more sources that provide suggestions. These sources are iframes into other apps, provied by thee intent system.
+    // Since we need to call the sources whenever the query changes, we are taking manual control over the intent process.
+    fetchRawIntent(INTENT_VERB, INTENT_DOCTYPE)
+    .then(intent => {
+      const { services } = intent.attributes
+      if (!services) return null
+      
+      services.forEach(service => {
+        const url = service.href
+        this.setState(state => ({...state, sourceURLs: [...state.sourceURLs, url]}))
+        const serviceOrigin = url.split('/', 3).join('/')
+        
+        this.sources.push({
+          slug: service.slug,     // can be used to show where a suggestion comes from
+          origin: serviceOrigin,
+          id: intent._id,
+          ready: false,
+          window: null,           // will hold a reference to the window we're sending messages to
+          resolve: null           // a reference to a function to call when the source sends suggestions
+        })
+      })
```

Comment:
```js
this.sources = services.map(service => {
  const url = service.href
  this.setState((state) => ({...state, sourceURLs: [...state.sourceURLs, url]})
  const serviceOrigin = url.split('/', 3).join('/')
  return {
    slug: service.slug,
    origin: serviceOrigin,
    id: intent._id,
    ready: false,
    window: null,
    resolve: null
  }
})
```

That let you remove instantiation `this.sources = []` from the constructor.

That let you remove the constructor in favor of a class properties for setting `this.state`.